tests/ui/bytes_nth.rs

```diff
@@ -0,0 +1,9 @@
+// run-rustfix
+
+#![warn(clippy::bytes_nth)]
+
+fn main() {
+    let _ = "Hello".bytes().nth(3);
+
+    let _ = String::from("Hello").bytes().nth(3);
+}
```

Comment:
A few more tests would be nice:

```rust
let s = String::from("String");
s.bytes().nth(3);
&s.bytes().nth(3);
s[..].bytes().nth(3);
```

They should all be linted, I think?

Reply:
I've just added some test cases!